misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -371,27 +372,34 @@ private void createReport(final IBundleCoverage bundleCoverage, Map<String, Modu
                             if (missedLinesList.isPresent() && coveredLinesList.isPresent()) {
                                 List<Integer> missedLines = missedLinesList.get();
                                 List<Integer> coveredLines = coveredLinesList.get();
-                                List<Integer> existingMissedLines = new ArrayList<>(missedLines);
                                 boolean isCoverageUpdated = false;
-                                int updateMissedLineCount = 0;
-                                for (int missedLine : existingMissedLines) {
-                                    // Traverse through the missed lines of a source file and update
-                                    // coverage status if it is covered in the current module.
-                                    // This is to make sure multi module tests are reflected in test coverage
-                                    ILine line = sourceFileCoverage.getLine(missedLine);
-                                    if (line.getStatus() == PARTLY_COVERED || line.getStatus() == FULLY_COVERED) {
+                                int updatedCoveredLineCount = 0;
+                                int updatedMissedLineCount = 0;
```

Comment:
When we moved to type values and typedesc to separate packages, we need to combine coverage for multiple packges with same source file.